appserver/web/web-glue/src/main/java/com/sun/enterprise/web/VirtualServer.java

```diff
@@ -621,7 +620,8 @@ public WebModuleConfig createSystemDefaultWebModuleIfNecessary(WebArchivist webA
             WebBundleDescriptorImpl webBundleDescriptor = webArchivist.getDefaultWebXmlBundleDescriptor();
             webBundleDescriptor.setModuleID(DEFAULT_WEB_MODULE_NAME);
             webBundleDescriptor.setContextRoot("");
-            
+            webBundleDescriptor.getLoginConfiguration().setRealmName(DEFAULT_WEB_MODULE_REALM);
+
```

Comment:
`file` is not always a good default. I think the correct default is whatever is specified in `SecurityService#getDefaultRealm` of current server's config.